package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -991,23 +988,9 @@ export const MessageInputProvider = <
     let response = {} as SendFileAPIResponse;
 
     try {
-      /**
-       * Expo now uses the assets-library which is also how remote
-       * native files are presented. We now return a file id from Expo
-       * only, if that file id exits we call getLocalAssetUri to download
-       * the asset for expo before uploading it. We do the same for native
-       * if the uri includes assets-library, this uses the CameraRoll.save
-       * function to also create a local uri.
-       */
-      const getLocalUri = async () => {
-        if (file.id) {
-          return await getLocalAssetUri(file.id);
-        } else if (file.uri?.match(/assets-library/)) {
-          return await getLocalAssetUri(file.uri);
-        }
-        return file.uri;
-      };
-      const uri = file.name || (await getLocalUri()) || '';
+      // For the case of expo messaging app where you need to fetch the file uri from file id. Here it is only done for iOS since for android the file.uri is fine.
+      const localAssetURI = Platform.OS === 'ios' && file.id && (await getLocalAssetUri(file.id));
```

Comment:
Can we incorporate this logic in `getLocalAssetUri` function? because DRY

Reply:
`getLocalAssetUri` is a native util. I don't think it makes sense to introduce such an logic inside it

Reply:
👍🏻